src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/AzureWorkloadProviderHelper.cs

```diff
@@ -62,6 +64,35 @@ public void RefreshContainer(string vaultName = null, string resourceGroupName =
             }
         }
 
+        public void RegisterContainer(string storageAccountName, ProtectionContainerResource res,
+           string vaultName, string resourceGroupName, string containerName)
+        {
+            string errorMessage = string.Empty;
+            var registerResponse = ServiceClientAdapter.RegisterContainer(
+                            storageAccountName,
+                            res,
+                            vaultName,
+                            resourceGroupName);
+
+            var operationStatus = TrackingHelpers.GetOperationResult(
+                registerResponse,
+                operationId =>
+                    ServiceClientAdapter.GetRegisterContainerOperationResult(
+                        operationId,
+                        containerName,
+                        vaultName: vaultName,
+                        resourceGroupName: resourceGroupName));
+
+            //Now wait for the operation to Complete
+            if (registerResponse.Response.StatusCode
+                    != SystemNet.HttpStatusCode.NoContent)
+            {
+                errorMessage = string.Format(Resources.DiscoveryFailureErrorCode,
```

Comment:
Error message should be RegisterFailureErrorCode. Please check with PM for the correct text to use.